ucb-rit/coldfront
# Add support for multiple email addresses

Fixes #15

**Changes**
- Added an `EmailAddress` model.
- Added an authentication backend to allow users to log in with any verified `EmailAddress`.
- Added views for adding, verifying, and removing `EmailAddress` objects from one's account, as well as a view for setting one as the account's primary address.
- Added a management command for creating `EmailAddress` objects for existing users that have none.
- Added help text to various places.

**How to Test**
- Manual Testing
    - Copy `local_settings.py.sample` to `local_settings.py`, or just add the variables under the new "Multiple Email Address settings" section.
    - Copy `local_strings.py.sample` to `local_strings.py`.
    - Run migrations: `python manage.py migrate`.
    - Create `EmailAddress` objects for existing users: `python manage.py create_email_addresses`.
    - Test that registration checks against existing `EmailAddress` objects.
    - Test that a user can authenticate with any verified email address.
    - Test that the User Profile functionality (adding, verifying, removing, setting primary) behave as expected.
